### PR TITLE
Clean up pedestrian class

### PR DESCRIPTION
**Pull request by schroedtert**
Removes following from the pedestrian class:
- not implemented
- not called functions
- not used variables

Minor fixes:
- Fixes most clang-tidy warnings
- renaming of variables to fit code convention (not raising clang-tidy warnings anymore) 

Subtask of #616

### CODE REVIEW

File: libcore/src/pedestrian/Pedestrian.cpp [hunk 1]
```diff
@@ -1101,16 +815,17 @@ int Pedestrian::GetColor() const
             double v0 = GetV0Norm();
             if(v0 != 0.0) {
                 double v = GetV().Norm();
-                color    = (int) (v / v0 * 255);
+                color    = static_cast<int>(v / v0 * 255);
             }
             return color;
-        } break;
```
Comment (JetteSchumann) [inline libcore/src/pedestrian/Pedestrian.cpp:1107]: you have removed a `break` here. this didn't change the logical part?

File: libcore/src/pedestrian/Pedestrian.cpp [hunk 2]
```diff
@@ -1123,7 +838,7 @@ int Pedestrian::GetColor() const
                                           // colors clearly distinguishable form
                                           // each other
             return (colors[_group % colors.size()]);
-        } break;
```
Comment (JetteSchumann) [inline libcore/src/pedestrian/Pedestrian.cpp:1126]: same here

File: libcore/src/pedestrian/Pedestrian.cpp (hunk 1 above)
Comment (schroedtert) [reply libcore/src/pedestrian/Pedestrian.cpp:1107]: no, as it is unreachable code. Due to the `return` before the function ends before the `break` can be reached.

File: libcore/src/pedestrian/Pedestrian.cpp (hunk 2 above)
Comment (schroedtert) [reply libcore/src/pedestrian/Pedestrian.cpp:1126]: see above

**Pull request merged**